### PR TITLE
feat: adding integ test to workflow

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
N/A

Description of changes:
Adding an integration test runner job for GitHub PR workflow.

Checklist:
[x] Have you successfully deployed to an AWS account with your changes?

AS review ticket id: N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/deploy-integ.yml [hunk 1]
```diff
@@ -48,6 +48,32 @@ jobs:
...
+          npm install -g pnpm
+          sudo apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Run integration tests
+        run: ./scripts/run-integration-tests.sh ${STAGE_NAME}
```
Comment (Bingjiling) [inline .github/workflows/deploy-integ.yml:67]: Where is STAGE_NAME defined and where is the STAGE_NAME.yml file?

Comment (SanketD92) [reply .github/workflows/deploy-integ.yml:67]: Added the env variable now. And I've manually placed the STAGE_NAME.yml file in the S3 bucket for now.

File: .github/workflows/deploy-integ.yml [hunk 2]
```diff
@@ -48,6 +48,33 @@ jobs:
...
+        uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - name: Install pnpm and system libraries
+        run: |
+          npm install -g pnpm
+          sudo apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+      - name: Configure AWS credentials
```
Comment (nguyen102) [inline .github/workflows/deploy-integ.yml]: I don't think we need this, we could set them up as env variables similar to line 44 to 47. I figure we stay consistent and either set up the creds with this GH action or using env vars.

File: .github/workflows/deploy-integ.yml [hunk 3]
```diff
@@ -48,6 +48,33 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          STAGE_NAME: e2etest
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Run integration tests
+        run: ./scripts/run-integration-tests.sh ${STAGE_NAME}
+        working-directory: ./
+        env:
```
Comment (nguyen102) [inline .github/workflows/deploy-integ.yml:69]: Do we need these vars as well?
INT_TEST_DIR
ENV_NAME

https://github.com/SanketD92/service-workbench-on-aws/blob/feat_GH_integ_workflow/scripts/run-integration-tests.sh

Comment (SanketD92) [reply .github/workflows/deploy-integ.yml:69]: ENV_NAME is the argument passed to this script, whereas INT_TEST_DIR would be initialized as part of sourcing util.sh script (line 8 in run-integration-tests.sh)

File: .github/workflows/deploy-integ.yml [hunk 4]
```diff
@@ -48,6 +48,30 @@ jobs:
...
+          sudo apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+      - name: Run integration tests
+        run: ./scripts/run-integration-tests.sh ${STAGE_NAME}
+        working-directory: ./
+        env:
+          DEPLOYMENT_BUCKET: ${{ secrets.DEPLOYMENT_BUCKET}}
+          STAGE_NAME: e2etest
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
```
Comment (nguyen102) [inline .github/workflows/deploy-integ.yml:74]: On line 78, change `needs: integration-test`. Let's run the tests in series for now incase there are any conflicts between integration tests and e2e tests.

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**